added missing header files for size_t and off_t.

jonmayergoogle opened the pull request:
Google's build tools are a little more strict than others.  To make our build work, we had to add these two missing header file dependencies.

LanceThompson commented:
Jon - could you elaborate on the operating system level and tool chain that you use that showed this as an error or warning?  We addressed this earlier and are a bit surprised that you are seeing an error.  We had a sample app.c and by changing the order of the includes there, we fixed it.

jonmayergoogle commented:
Hi Lance,

We're building on linux using LLVM-derived compiler, which tends to be a
little more strict than gcc.  I'm building the as a library, and without
this change we get parser errors for individual compilation units.

<philosophy>Google's coding style believes that "include what you use" is
the right policy.  You can sometimes mask issues like this by ordering
#include statements in a particular way, but include-what-you-use means
that your code will always compile regardless of how the includes are
ordered, so we prefer it.</philosophy>

Thanks!

- Jonathan.


On Tue, Jul 19, 2016 at 8:12 AM, Lance Thompson <notifications@github.com>
wrote:

> Jon - could you elaborate on the operating system level and tool chain
> that you use that showed this as an error or warning? We addressed this
> earlier and are a bit surprised that you are seeing an error. We had a
> sample app.c and by changing the order of the includes there, we fixed it.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ibm-capi/pslse/pull/50#issuecomment-233664694>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ATgCPBOqJDqLoyftmW820fT6s4O7X9EJks5qXOl0gaJpZM4JPC19>
> .
>



-- 
Jonathan Mayer
email:jonmayer@google.com cell:650-787-8235
